mitmproxy/addons/maplocal.py

```diff
@@ -0,0 +1,104 @@
+import mimetypes
+import re
+import typing
+import urllib
+from pathlib import Path
+from werkzeug.security import safe_join
+
+from mitmproxy import ctx, exceptions, http
+from mitmproxy.addons.modifyheaders import parse_modify_spec, ModifySpec
+
+
+def get_mime_type(file_path: str) -> str:
+    mimetype = (
+        mimetypes.guess_type(file_path)[0]
+        or "application/octet-stream"
+    )
+    return mimetype
+
+
+def file_candidates(url: str, base_path: str) -> typing.List[Path]:
+    candidates = []
+    parsed_url = urllib.parse.urlparse(url)
+    path_components = parsed_url.path.lstrip("/").split("/")
```

Comment:
If we want to operate on path components, there's already `flow.request.path_components`. Intuitively I feel that this may work better matching on substrings, but if we get tests passing with this, we're good of course. :)

Reply:
Good idea, will change it.